### PR TITLE
lib: For MultiBalanceReport, report change in valuation rather than valuation of change.

### PR DESCRIPTION
**Pull request by Xitian9**
Addresses the MultiBalanceReport part of #1353

### CODE REVIEW

**Comment by Xitian9** [comment]
The history of this PR seems to have been contaminated a bit. I'll rebase this when we've sorted out #1354.

**Comment by Xitian9** [comment]
Rebased.

**Comment by simonmichael** [comment]
Four tests changed, but their descriptions didn't change - are those still correct ?

I guess we'll need changes somewhere under https://hledger.org/hledger.html#valuation - at https://hledger.org/hledger.html#effect-of-valuation-on-reports and perhaps somewhere above.

**Comment by simonmichael** [comment]
Looking good, I confirm that it fixes @charvp's bs example too.

Perhaps the test descriptions, and report headings, should change something like this ?
```
# 35. multicolumn balance report valued at period end
$ hledger -f- bal -M --value=end
Balance changes in 2000Q1, valued at period ends:
```
->
```
# 35. multicolumn balance report showing changes in period-end values
$ hledger -f- bal -M --value=end
Period-end value changes in 2000Q1:
```

**Comment by Xitian9** [comment]
That change in doc/description sounds good to me.

**Comment by Xitian9** [comment]
Hmm. Thinking about the doc/heading a bit more. Should we distinguish between cases where it's checking the change in valued  balance, and cases where it's valuing the change? The latter happens whenever there's no valuation, or when valuation happens at a fixed date for all columns.

**Comment by simonmichael** [comment]
Yes it sounds like we should, probably. For each report we support, the title should be something that helps users keep straight on what the numbers mean.

**Comment by simonmichael** [comment]
For the docs: it looks like only the "Effect of valuation on reports" table needs updating. Maybe we should split the "balances (with report interval)" row into "end balances (with report interval)" and "balance changes (with report interval)" rows.

**Comment by Xitian9** [comment]
Alright, I did my best. If you can think of a more succinct way of describing the Effect on valuation reports, feel free to change it.

File: tests/journal/valuation.test [hunk 1]
```diff
@@ -547,7 +547,7 @@ Budget performance in 2000Q1, valued at cost:
 ---++------------------------------------------------------------------------------------------------
    || 6 B [300% of 2 B]  7 B [350% of 2 B]  8 B [400% of 2 B]  21 B [350% of 6 B]  7 B [350% of 2 B] 
 
-# 48. budget report, valued at period ends.
+# 48. budget report, showing changes in period-end values.
 $ hledger -f- bal -MTA --budget --value=e
 Budget performance in 2000Q1, valued at period ends:
```
Comment (simonmichael) [inline tests/journal/valuation.test:552]: I can add this as a fixup, but need a sanity check: the budget report will also show value changes, and we can title it "Budget performance in 2000Q1, period-end value changes:"  ?

Comment (simonmichael) [reply tests/journal/valuation.test:552]: Oh I see, BudgetReport is separate and has not changed. 

Looks good to me, thank you!

**Comment by simonmichael** [comment]
https://hledger.org/hledger.html#effect-of-valuation-on-reports tweaked

**Pull request merged**